scripts/lua2dox.lua

```diff
@@ -203,7 +203,14 @@ local function process_magic(line, generics)
     for _, type in ipairs(TYPES) do
       magic = magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. ')%)', 'param %1 %2')
       magic =
-        magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '|nil)%)', 'param %1 %2')
+          magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '|nil)%)', 'param %1 %2')
+      magic =
+          magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '%?)%)', 'param %1 %2')
+      magic = magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '%[%])%)', 'param %1 %2')
+      magic =
+          magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '%[%]|nil)%)', 'param %1 %2')
+      magic =
+          magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '%[%]%?|nil)%)', 'param %1 %2')
```

Comment:
Is there not a simpler and more robust way to handle this rather than matching every possibility?

Reply:
There is. I implemented a pattern that combines these cases.